src/rust/ide/src/controller/graph.rs

```diff
@@ -0,0 +1,218 @@
+//! Graph Controller.
+//!
+//! This controller provides access to a specific graph. It lives under a module controller, as
+//! each graph belongs to some module.
+
+
+use crate::prelude::*;
+
+pub use crate::double_representation::graph::Id;
+
+use flo_stream::MessagePublisher;
+use flo_stream::Subscriber;
+use utils::channel::process_stream_with_handle;
+
+
+
+// ==============
+// === Errors ===
+// ==============
+
+/// Error raised when node with given Id was not found in the graph's body.
+#[derive(Clone,Copy,Debug,Fail)]
+#[fail(display="Node with Id {} was not found.", _0)]
+pub struct NodeNotFound(ast::ID);
+
+
+
+// ============
+// === Node ===
+// ============
+
+/// TODO: replace with usage of the structure to be provided by Josef
+#[derive(Clone,Copy,Debug)]
+pub struct NodeMetadata; // here goes position
+
+/// Error raised when node with given Id was not found in the graph's body.
+#[derive(Clone,Debug)]
+pub struct Node {
+    /// Information based on AST, from double_representation module.
+    pub info : double_representation::node::NodeInfo,
+    /// Information about this node stored in the module's metadata.
+    pub metadata : NodeMetadata
+}
+
+
+
+// ===================
+// === NewNodeInfo ===
+// ===================
+
+/// Describes the node to be added.
+#[derive(Clone,Debug)]
+pub struct NewNodeInfo {
+    /// Expression to be placed on the node
+    pub expression : String,
+    /// Visual node position in the graph scene.
+    pub metadata : NodeMetadata,
```

Comment:
Shan't we rather keep position in some "Position" structure? As I understand, this is structure used in API, and fact that the position is kept in metadata looks like our internal solution.